### ISSUE TITLE
DB error (OperationalError) after removing and renaming several endpoints. 

### ISSUE THREAD

**Issue by mircealungu** [flask-dashboard/Flask-MonitoringDashboard#54]
I've removed several endpoints and also renamed several.
when i redeployed the API i had no errors. 
however, later, when accessing the renamed endpoint
( i remember for sure that it happened when i accessed 
 'stop_following_feed' which occurs also in this trace) 
an exception in the Dashboard is thrown which kills the
entire endpoint which results in a 500 error. 

Great would be to: 
- make sure the exception is not thrown
Better than nothing would be to: 
- catch such exceptions and let the endpoint do it's 
job even if the tracking crashes. 

To fix this I've deployed the API with dasboardEnabled=False
once. Then re-enabled it and it worked. 

I can't reproduce, because now it works. 

But this might be useful for you. 

[Wed Feb 21 15:24:15.754589 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61859]     compiled_sql, distilled_params
[Wed Feb 21 15:24:15.754605 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61859]   File "/usr/local/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1189, in _execute_context
[Wed Feb 21 15:24:15.754613 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61859]     context)
[Wed Feb 21 15:24:15.754628 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61859]   File "/usr/local/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1402, in _handle_dbapi_exception
[Wed Feb 21 15:24:15.754637 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61859]     exc_info
[Wed Feb 21 15:24:15.754652 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61859]   File "/usr/local/lib/python3.6/site-packages/sqlalchemy/util/compat.py", line 203, in raise_from_cause
[Wed Feb 21 15:24:15.754660 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61859]     reraise(type(exception), exception, tb=exc_tb, cause=cause)
[Wed Feb 21 15:24:15.754676 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61859]   File "/usr/local/lib/python3.6/site-packages/sqlalchemy/util/compat.py", line 186, in reraise
[Wed Feb 21 15:24:15.754685 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61859]     raise value.with_traceback(tb)
[Wed Feb 21 15:24:15.754700 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61859]   File "/usr/local/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1182, in _execute_context
[Wed Feb 21 15:24:15.754708 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61859]     context)
[Wed Feb 21 15:24:15.754723 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61859]   File "/usr/local/lib/python3.6/site-packages/sqlalchemy/engine/default.py", line 470, in do_execute
[Wed Feb 21 15:24:15.754732 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61859]     cursor.execute(statement, parameters)
[Wed Feb 21 15:24:15.754759 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61859] sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) database is locked [SQL: 'UPDATE rules SET last_accessed=? WHERE rules.endpoint = ?'] [parameters: ('2018-02-21 15:24:10.741125', 'api.get_non_subscribed_feeds')]
[Wed Feb 21 15:24:22.525761 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866] mod_wsgi (pid=3664): Exception occurred processing WSGI script '/home/mircea/zee/http/api/zeeguu_api.wsgi'.
[Wed Feb 21 15:24:22.526359 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866] Traceback (most recent call last):
[Wed Feb 21 15:24:22.526402 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]   File "/usr/local/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1182, in _execute_context
[Wed Feb 21 15:24:22.526412 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]     context)
[Wed Feb 21 15:24:22.526448 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]   File "/usr/local/lib/python3.6/site-packages/sqlalchemy/engine/default.py", line 470, in do_execute
[Wed Feb 21 15:24:22.526457 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]     cursor.execute(statement, parameters)
[Wed Feb 21 15:24:22.526488 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866] sqlite3.OperationalError: database is locked
[Wed Feb 21 15:24:22.526512 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]
[Wed Feb 21 15:24:22.526521 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866] The above exception was the direct cause of the following exception:
[Wed Feb 21 15:24:22.526528 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]
[Wed Feb 21 15:24:22.526543 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866] Traceback (most recent call last):
[Wed Feb 21 15:24:22.526705 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]   File "/usr/local/lib/python3.6/site-packages/Flask-0.12.2-py3.6.egg/flask/app.py", line 1997, in __call__
[Wed Feb 21 15:24:22.526714 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]     return self.wsgi_app(environ, start_response)
[Wed Feb 21 15:24:22.526730 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]   File "/usr/local/lib/python3.6/site-packages/Flask-0.12.2-py3.6.egg/flask/app.py", line 1985, in wsgi_app

[Wed Feb 21 15:24:22.526739 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]     response = self.handle_exception(e)
[Wed Feb 21 15:24:22.526755 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]   File "/usr/local/lib/python3.6/site-packages/Flask_Cors-3.0.2-py3.6.egg/flask_cors/extension.py", line 161, in wrapped_function
[Wed Feb 21 15:24:22.526764 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]     return cors_after_request(app.make_response(f(*args, **kwargs)))
[Wed Feb 21 15:24:22.526780 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]   File "/usr/local/lib/python3.6/site-packages/Flask-0.12.2-py3.6.egg/flask/app.py", line 1540, in handle_exception
[Wed Feb 21 15:24:22.526789 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]     reraise(exc_type, exc_value, tb)
[Wed Feb 21 15:24:22.526804 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]   File "/usr/local/lib/python3.6/site-packages/Flask-0.12.2-py3.6.egg/flask/_compat.py", line 33, in reraise
[Wed Feb 21 15:24:22.526812 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]     raise value
[Wed Feb 21 15:24:22.526828 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]   File "/usr/local/lib/python3.6/site-packages/Flask-0.12.2-py3.6.egg/flask/app.py", line 1982, in wsgi_app
[Wed Feb 21 15:24:22.526836 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]     response = self.full_dispatch_request()
[Wed Feb 21 15:24:22.526852 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]   File "/usr/local/lib/python3.6/site-packages/Flask-0.12.2-py3.6.egg/flask/app.py", line 1614, in full_dispatch_request
[Wed Feb 21 15:24:22.526861 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]     rv = self.handle_user_exception(e)
[Wed Feb 21 15:24:22.526876 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]   File "/usr/local/lib/python3.6/site-packages/Flask_Cors-3.0.2-py3.6.egg/flask_cors/extension.py", line 161, in wrapped_function
[Wed Feb 21 15:24:22.526885 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]     return cors_after_request(app.make_response(f(*args, **kwargs)))
[Wed Feb 21 15:24:22.526907 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]   File "/usr/local/lib/python3.6/site-packages/Flask-0.12.2-py3.6.egg/flask/app.py", line 1517, in handle_user_exception
[Wed Feb 21 15:24:22.526916 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]     reraise(exc_type, exc_value, tb)
[Wed Feb 21 15:24:22.526932 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]   File "/usr/local/lib/python3.6/site-packages/Flask-0.12.2-py3.6.egg/flask/_compat.py", line 33, in reraise
[Wed Feb 21 15:24:22.526940 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]     raise value
[Wed Feb 21 15:24:22.526955 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]   File "/usr/local/lib/python3.6/site-packages/Flask-0.12.2-py3.6.egg/flask/app.py", line 1612, in full_dispatch_request
[Wed Feb 21 15:24:22.526964 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]     rv = self.dispatch_request()
[Wed Feb 21 15:24:22.526979 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]   File "/usr/local/lib/python3.6/site-packages/Flask-0.12.2-py3.6.egg/flask/app.py", line 1598, in dispatch_request
[Wed Feb 21 15:24:22.526988 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]     return self.view_functions[rule.endpoint](**req.view_args)
[Wed Feb 21 15:24:22.527004 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]   File "/usr/local/lib/python3.6/site-packages/flask_monitoring_dashboard-1.9-py3.6.egg/dashboard/measurement.py", line 86, in wrapper
[Wed Feb 21 15:24:22.527013 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]     update_last_accessed(endpoint=endpoint, value=datetime.datetime.now())
[Wed Feb 21 15:24:22.527029 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]   File "/usr/local/lib/python3.6/site-packages/flask_monitoring_dashboard-1.9-py3.6.egg/dashboard/database/endpoint.py", line 132, in update_last_accessed
[Wed Feb 21 15:24:22.527038 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]     update({MonitorRule.last_accessed: value})
[Wed Feb 21 15:24:22.527053 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]   File "/usr/local/lib/python3.6/site-packages/sqlalchemy/orm/query.py", line 3345, in update

[Wed Feb 21 15:24:22.527062 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]     update_op.exec_()
[Wed Feb 21 15:24:22.527077 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]   File "/usr/local/lib/python3.6/site-packages/sqlalchemy/orm/persistence.py", line 1179, in exec_
[Wed Feb 21 15:24:22.527085 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]     self._do_exec()
[Wed Feb 21 15:24:22.527100 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]   File "/usr/local/lib/python3.6/site-packages/sqlalchemy/orm/persistence.py", line 1334, in _do_exec
[Wed Feb 21 15:24:22.527109 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]     mapper=self.mapper)
[Wed Feb 21 15:24:22.527124 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]   File "/usr/local/lib/python3.6/site-packages/sqlalchemy/orm/session.py", line 1139, in execute
[Wed Feb 21 15:24:22.527132 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]     bind, close_with_result=True).execute(clause, params or {})
[Wed Feb 21 15:24:22.527148 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]   File "/usr/local/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 945, in execute
[Wed Feb 21 15:24:22.527156 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]     return meth(self, multiparams, params)
[Wed Feb 21 15:24:22.527172 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]   File "/usr/local/lib/python3.6/site-packages/sqlalchemy/sql/elements.py", line 263, in _execute_on_connection
[Wed Feb 21 15:24:22.527186 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]     return connection._execute_clauseelement(self, multiparams, params)
[Wed Feb 21 15:24:22.527203 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]   File "/usr/local/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1053, in _execute_clauseelement
[Wed Feb 21 15:24:22.527211 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]     compiled_sql, distilled_params
[Wed Feb 21 15:24:22.527226 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]   File "/usr/local/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1189, in _execute_context
[Wed Feb 21 15:24:22.527235 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]     context)
[Wed Feb 21 15:24:22.527250 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]   File "/usr/local/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1402, in _handle_dbapi_exception
[Wed Feb 21 15:24:22.527258 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]     exc_info
[Wed Feb 21 15:24:22.527274 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]   File "/usr/local/lib/python3.6/site-packages/sqlalchemy/util/compat.py", line 203, in raise_from_cause
[Wed Feb 21 15:24:22.527282 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]     reraise(type(exception), exception, tb=exc_tb, cause=cause)
[Wed Feb 21 15:24:22.527298 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]   File "/usr/local/lib/python3.6/site-packages/sqlalchemy/util/compat.py", line 186, in reraise
[Wed Feb 21 15:24:22.527307 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]     raise value.with_traceback(tb)
[Wed Feb 21 15:24:22.527322 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]   File "/usr/local/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1182, in _execute_context
[Wed Feb 21 15:24:22.527330 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]     context)
[Wed Feb 21 15:24:22.527345 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]   File "/usr/local/lib/python3.6/site-packages/sqlalchemy/engine/default.py", line 470, in do_execute
[Wed Feb 21 15:24:22.527354 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866]     cursor.execute(statement, parameters)
[Wed Feb 21 15:24:22.527381 2018] [wsgi:error] [pid 3664:tid 140584959235840] [remote 94.214.187.58:61866] sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) database is locked [SQL: 'UPDATE rules SET last_accessed=? WHERE rules.endpoint = ?'] [parameters: ('2018-02-21 15:24:17.520847', 'api.stop_following_feed')]

**Comment by FlyingBird95** [comment]
Interested link: https://stackoverflow.com/questions/13895176/sqlalchemy-and-sqlite-database-is-locked

**Comment by kloostert** [comment]
I also tried to reproduce this, but in the newest version I do not get this error.
I did add some code in our wrapper functions that the Dashboard uses for tracking the endpoints.
This added code (#112) ensures that endpoints that have been called can still function properly, even when the Dashboard tracking fails for some reason. 
Any errors that the Dashboard may encounter will still be printed, along with a stack trace, the only difference is that these errors will not kill an endpoint anymore.
